### PR TITLE
Support cross device rename for helm v3

### PR DESCRIPTION
**Pull request by ysh7**
This PR includes changes on #6738 for Helm v3

/cc @thomastaylor312

### CODE REVIEW

File: pkg/downloader/manager.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ import (
 	"github.com/pkg/errors"
 	"sigs.k8s.io/yaml"
 
+	"helm.sh/helm/v3/internal/dep/fs"
```
Comment (thomastaylor312) [inline pkg/downloader/manager.go]: This should be 
```suggestion
	"helm.sh/helm/v3/internal/third_party/dep/fs"
```

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Thanks for opening the port @ysh7! One error in here that is causing the build failure

File: pkg/downloader/manager.go (hunk 1 above)
Comment (ysh7) [reply pkg/downloader/manager.go]: Change you asked for is done. Sorry this was very lame mistake on my part.

**Comment by thomastaylor312** [comment]
@ysh7 Thanks for fixing that! Looks like you pulled in some commits that were already in master. Anyway you could rebase to remove those?

**Review by thomastaylor312 — APPROVED** [review]
This LGTM from my end. Pinging @mattfarina for a second review on this

**Comment by thomastaylor312** [comment]
This slipped by our review of the queue for some reason. Pulling in to 3.0 as we are also pulling #6738 into 2.16

**Review by mattfarina — APPROVED** [review]

**Pull request merged**